### PR TITLE
Changed "Duplicate Identifier" to "enum cannot be merged..."

### PR DESCRIPTION
**Pull request by NaridaL**
when either declaration of the identifier is an enum.

Partial (?) fix for #529

### CODE REVIEW

**Comment by mhegazy** [comment]
@DanielRosenwasser can i get your input on the proposed error message here?

**Comment by NaridaL** [comment]
@DanielRosenwasser If you OK the message, I'll rebase the PR and we can merge it^^.

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -1924,6 +1924,11 @@
         "category": "Error",
         "code": 2563
     },
+    "An enum cannot be merged with a class, interface, function, namespace or variable.": {
```
Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: how about `Enum declarations can only merge with namespace or other enum declarations.`

**Comment by NaridaL** [comment]
@mhegazy "Enum declarations can only merge with namespace or other enum declarations." definitely seems better. I rebased and changed it.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -847,8 +847,13 @@ namespace ts {
                 error(getNameOfDeclaration(source.declarations[0]), Diagnostics.Cannot_augment_module_0_with_value_exports_because_it_resolves_to_a_non_module_entity, symbolToString(target));
             }
             else {
-                const message = target.flags & SymbolFlags.BlockScopedVariable || source.flags & SymbolFlags.BlockScopedVariable
-                    ? Diagnostics.Cannot_redeclare_block_scoped_variable_0 : Diagnostics.Duplicate_identifier_0;
+                let message = Diagnostics.Duplicate_identifier_0;
```
Comment (mhegazy) [inline src/compiler/checker.ts]: nit:

```ts
const message = target.flags & SymbolFlags.Enum || source.flags & SymbolFlags.Enum
    ? Diagnostics.Enum_declarations_can_only_merge_with_namespace_or_other_enum_declaration
    : target.flags & SymbolFlags.BlockScopedVariable || source.flags & SymbolFlags.BlockScopedVariable
        ? Diagnostics.Cannot_redeclare_block_scoped_variable_0
        : Diagnostics.Duplicate_identifier_0;
```

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**